resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -532,7 +533,8 @@ class ExecutorPodsAllocator(
       currentTime - creationTime > executorIdleTimeout
     } catch {
       case e: Exception =>
-        logError(s"Cannot get the creationTimestamp of the pod: ${state.pod}", e)
+        logError(log"Cannot get the creationTimestamp of the pod: " +
+          log"${MDC(LogKey.POD_ID, state.pod)}", e)
```

Comment:
IMO `POD_ID` is not a suitable key here. `state.pod` actually outputs a whole Pod Spec, and there is no such a "pod id" concept in the K8s, in practice, we use `namespace` and `pod name` to identify a pod.

Reply:
@pan3793 Thanks for the suggestion. Do you mind creating a PR to improve this?

Reply:
I would like to have a try, it may take a few hours to learn this new log framework.

Reply:
Thank you in advance!

Reply:
@pan3793 Thank you very much for `correcting` this issue.